---
title: "GO Term Enrichment"
author: "Noelle Wheeler"
date: "2023-03-09"
output: html_document
---

load datasets and libraries
```{r, warning=FALSE, results = 'hide', message=FALSE}
# BiocManager::install("DESeq2")
library(DESeq2)
library(magrittr)
FILE_DSD="/Users/noelawheeler/Downloads/RNAseqGierlinski.RData"
load(FILE_DSD)
DESeq.ds$condition %<>% relevel(ref="WT")
design(DESeq.ds)
DESeq.ds %<>% DESeq()
DESeq.ds
```

```{r}
# plot raw p-values
rowData(DESeq.ds)$WaldPvalue_condition_SNF2_vs_WT %>%
hist(breaks=19, main="Raw p-values for SNF2 vs WT")

# filter results with multiple testing correction and remove lowly expressed genes  
DGE.results <- results(DESeq.ds, independentFiltering = TRUE, alpha = 0.05)

# plot adjusted p-values
DGE.results$padj %>%
hist(breaks=19, main="Adjusted p-values for SNF2 vs WT")

# find genes with target p-value
DGE.genes <- rownames(subset(DGE.results, padj < 0.05))
```


### 1. Using the `pwf` table, after what length of gene does the bias appear to stabilize? 
```{r, message = FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

# install org.Sc.sgd.db annotation package
# BiocManager::install("org.Sc.sgd.db", force = TRUE)
library(org.Sc.sgd.db)
sacCer <- org.Sc.sgd.db

# install GOSeq
# BiocManager::install("goseq", force = TRUE)
library(goseq)

# install sacCer3.sgdGene
# BiocManager::install("TxDb.Scerevisiae.UCSC.sacCer3.sgdGene", force = TRUE)
library("TxDb.Scerevisiae.UCSC.sacCer3.sgdGene")
gene.vector <- row.names(DGE.results) %in% DGE.genes %>% as.integer
names(gene.vector) <- row.names(DGE.results)
pwf <- nullp(gene.vector, "sacCer3", "ensGene")
```

It appears that before a length of 2000 the bias is stable. 


### 2. What proportion of genes are below this length?
```{r}
length(which(pwf$pwf < .47)) / length(pwf$pwf)
```

### 3. For each of the ontologies (BP, CC, and MF), list the top 5 most overrepresented GO terms in your DE genes.
```{r}
GO.wall <- goseq(pwf, "sacCer3", "ensGene")
dim(GO.wall)
# head(GO.wall)
# separate by BP, CC, MF and list lowest 5 overrepresented GO terms
BP_values <- GO.wall[which(GO.wall$ontology == 'BP'),]
print(paste0("5 top overrepresented GO terms for BP: ", BP_values[seq(1:5), 1]))
# BP_values[seq(1:5), 1]
CC_values <- GO.wall[which(GO.wall$ontology == 'CC'),]
print(paste0("5 top overrepresented GO terms for CC: ", CC_values[seq(1:5), 1]))
# CC_values[seq(1:5), 1]
MF_values <- GO.wall[which(GO.wall$ontology == 'MF'),]
print(paste0("5 top overrepresented GO terms for MF: ", MF_values[seq(1:5), 1] ))
# MF_values[seq(1:5), 1]
```
### 4. Pick one of those categories and include in your write-up the treemap for that category, generated by the R code provided by REVIGO. 
```{r}
# focusing on BP
# get GO terms and p-value for BP ontology
all_BP <- BP_values[seq(1:5), c(1,2)]
write.table(all_BP,
    file = "/Users/noelawheeler/Desktop/Analysis of next gen seq data/Analysis-Next-Gen-Sequencing/BP_GOterms_goseq.txt",
    quote = FALSE, row.names = FALSE, col.names = FALSE)
```

![](/Users/noelawheeler/Desktop/Analysis of next gen seq data/BP_treemap.png)

### 5. Pick one of those categories, and identify a gene from that category that has differential expression. Use `plotCounts` to show the gene's counts across the two conditions.
```{r}
# retrieving the GO categories assigned to each gene for only DE 
go_genes_DE <- getgo(rownames(DGE.results[which(DGE.results$padj < 0.05),]), 'sacCer3', 'ensGene') %>% stack
# head(go_genes_DE)

# find a GO term for BP with lowest over_respresented_pvalue
GO_term <- all_BP[1, "category"]

# find all genes that have DE and have this GO term
BP_DE_genes <- go_genes_DE[which(go_genes_DE$values == GO_term),]

# look at first gene
BP_DE_gene <- BP_DE_genes[1,2]
```

```{r}
# plot
par(mfrow=c(1,2))
plotCounts(DESeq.ds, gene="YAL059W", normalized = TRUE, xlab="")
```



