---
title: "Exercise: Sequencing Data"
author: "Noelle Wheeler"
date: "2023-02-02"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Part I sequencing data
### Question 1

Download file with urls
```{}
cd homework
wget -O accession_url "https://www.ebi.ac.uk/ena/portal/api/filereport?accession=PRJEB5348&result=read_run&fields=run_accession,fastq_ftp&format=tsv&download=true&limit=0"
```

Create file to hold accession IDs
```{}
str=ERR
x=458878
while [ $x -le 458884 ]
do
    echo $str$x >> values.txt
    echo >> values.txt
    x=$(( $x + 1 ))
done
```

Download all related files and check they have downloaded

```{}
while read line
do
    wget -O $line.txt ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/$line/$line.fastq.gz
done < values.txt 

ls
```

### Question 2
They are there multiple fastq files for the sample WT_2? What does each file represent?

There are 7 fastq files for each biological replicate because each one corresponds to a different lane on a sequencing flow cell.

### Question 3
Count the number of lines in each fastq file
```{}
for file in ERR*; do zcat $file | wc -l; done
```

```
5870276
5800048
5766276
5286992
4527136
4562752
5846584
```

Count the number of reads in each fastq file
```{}
for file in ERR*; do count=$(zcat $file | wc -l); count_div=$(( $count / 4 )); echo $count_div; done
```

```
1093957
1078049
1066501
985619
846040
851243
1093200
1885330
```

## Part II sequencing quality control and preprocessing
### Question 1

Make directories for new files
```{}
mkdir SNF2_1 SNF2_2 SNF2_3 WT1 WT2 WT3
```

Download files so I have first three technical replicates of WT and SNF2 samples
```{}
#! bin/bash

str=ERR
x=$1

while [ $x -le $2 ]
do
    wget -O $str$x.txt ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR458/$str$x/$str$x.fastq.gz
    x=$(( $x + 1 ))
done
```

Pass in accession number ranges as arguments
```{}
./download.sh 458493 458520
./download.sh 458885 458891
```

### Question 2
Write a for loop that will run FastQC on all of the FASTQ files
```{}
for dir in SNF2_1 SNF2_2 SNF2_3 WT1 WT2 WT3
do
    for file in $dir/*
    do
        fastqc $file --extract
    done
done
```

### Question 2
Run TrimGalore on sample SNF2 biological replicate no. 1. I did not change any of the defualt parameters.
```{}
for file in SNF2_1/*fastq.gz
do
    trim_galore --illumina $file
done
```

### Question 2
Run FastQC on trimmed sample SNF2 biological replicate no. 1
```{}
for file in *trimmed.fq.gz
do
    fastqc $file --extract
done
```

### Question 3
Describe one detail of the QC results that changes after TrimGalore and one result that stays the same.

After TrimGalore there is a warning for Sequence Length Distribution. Some of the sequences are trimmed to be shorter than 51 bp whereas before trimming all the sequences are 51 bp. 
The sequence duplication levels appear similar between the trimmed and non trimmed sequences in the samples that I looked at. Trimming would not change the amount of duplicated sequences unless a duplicate sequence gets removed during the trimming process. 

### Question 4
Run MultiQC on initial FastQ results
```{}
mamba activate multiqc
multiqc SNF2_1/ SNF2_2/ SNF2_3/ WT1/ WT2/ WT3/
```
Here the WT samples are highlighted in red and the SNF2 samples are highlighted in green.
![](/Users/noelawheeler/Downloads/fastqc_sequence_counts_plot.png)

### Question 5
Based on the QC, would you be justified in combining any of the FASTQ files given that they are technical replicates?

I would not combine the technical replicates because they seem to have as much variation as the biological replicates. This may be due to batch effects that would be worth exploring and understanding.

### Question 6
What commands would you use to combine several FASTQ files into one?

```{}
for i in {3..9}
do
  cat ERR45849$i.fastq.gz >> WT.fastqc.gz
done

for i in {00..20}
do
  cat ERR4585$i.fastq.gz >> SNF2.fastqc.gz
done

for i in {78..91}
do
  cat ERR4588$i.fastq.gz >> WT.fastqc.gz
done
```
## Question 7
If you had to determine the version of the Sanger quality score encoding used in a given FASTQ file without the help of FastQC, what would you do?

I would consult the ascii table to see if the Phred scores seem to be +64 or +33, given that most quality scores will be around 30 or above. The +33 version will have more quality scores with uppercase letters and the +64 version will have more quality scores with lowercase letter. 
